### PR TITLE
Add json config and helm repo fetch support

### PR DESCRIPTION
**Pull request by mudler**
It adds support to config files and allows to directly fetch and re-assemble charts from the official helm repo.

Note: the `BACKEND` cannot yet be annotated in the json as it is needed in the Makefile and it will be ignored. And ENV vars have precedence over the one specified in the json (done on purpose!).

Json config:
`CONFIG=/path/to/json BACKEND=backend make [..]`

Use chart from helm (need to pass the SCF version):
`SCF_HELM_VERSION=2.14.5 make [..]`

e.g.:
```
$> cat support.json
{
  "SCF_HELM_VERSION": "2.14.5",
  "CLUSTER_NAME": "support",
  "ENABLE_EIRINI": "false",
  "EKCP_HOST": "....
}
```

### CODE REVIEW

File: modules/scf/chart.sh [hunk 1]
```diff
@@ -63,9 +63,11 @@ if [ -n "$SCF_HELM_VERSION" ]; then
     popd
 fi
 
+rm -rf chart_url || true
+
 if echo "$SCF_CHART" | grep -q "http"; then
-    info "Downloading $SCF_CHART"
     wget "$SCF_CHART" -O chart
+    echo "$SCF_CHART" > chart_url
```
Comment (viccuad) [inline modules/scf/chart.sh]: we should differentiate it from the stratos chart, maybe `scf_chart_url`?

File: include/buildir.sh [hunk 2]
```diff
@@ -1,11 +1,11 @@
 #!/bin/bash
 
 # duplicated in s/include/common.sh, needed for bootstrapping:
-. include/common.sh
+. $ROOT_DIR/include/common.sh
```
Comment (viccuad) [inline include/buildir.sh]: how does this work? isn't $ROOT_DIR undefined until one sources include/common.sh?

Comment (mudler) [reply include/buildir.sh]: It's in the top level of the Makefile, so it's always present :)

File: modules/scf/chart.sh (hunk 1 above)
Comment (mudler) [reply modules/scf/chart.sh]: Addressed in https://github.com/SUSE/catapult/pull/66/commits/1a970299781bc5b4c05f59f1f3a5ef347fb44f39

**Comment by viccuad** [comment]
I love this, many thanks! :smile:

**Pull request merged**